src/App.tsx

```diff
@@ -1,61 +1,115 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/types';
+
+const initialTodos: Todo[] = todosFromServer;
+
+function getNewTodoId(todos: Todo[]) {
+  const maxId = Math.max(0, ...todos.map(todo => todo.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>(initialTodos);
+
+  const addTodo = ({ id, ...data }: Todo) => {
+    const newTodo = {
+      id: getNewTodoId(todos),
+      ...data,
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  function handleTitle(e: React.ChangeEvent<HTMLInputElement>) {
+    setTitle(e.target.value);
+    setHasTitleError(false);
+  }
+
+  function handleUserId(e: React.ChangeEvent<HTMLSelectElement>) {
+    setUserId(+e.target.value);
+    setHasUserIdError(false);
+  }
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodoObj = {
+      id: 0,
```

Comment:
The `id` for new todos should be generated using the `getNewTodoId` function instead of hardcoding it to `0`. This will ensure each todo has a unique id.